include/zephyr/drivers/mspi.h

```diff
@@ -0,0 +1,808 @@
+/*
+ * Copyright (c) 2024, Ambiq Micro Inc. <www.ambiq.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for MSPI driver
+ */
+
+#ifndef ZEPHYR_INCLUDE_MSPI_H_
+#define ZEPHYR_INCLUDE_MSPI_H_
+
+#include <errno.h>
+
+#include <zephyr/sys/__assert.h>
+#include <zephyr/types.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief MSPI Driver APIs
+ * @defgroup mspi_interface MSPI Driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @brief MSPI operational mode
+ */
+enum mspi_op_mode {
+	MSPI_OP_MODE_MASTER         = 0,
+	MSPI_OP_MODE_SLAVE          = 1,
```

Comment:
introducing a new API that is using non inclusive language is a no-go IMO
https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-2-inclusive-language

Reply:
The SPI standard was written as MASTER/SLAVE.
The new terms are typically main/sub.

Where and how in Zephyr would we describe to a user that MAIN means MASTER and SUB means SLAVE and how would you do that without using the word, master and slave?

Reply:
> The new terms are typically main/sub.

So I guess

```suggestion
	MSPI_OP_MODE_MAIN         = 0,
	MSPI_OP_MODE_SUB          = 1,
```

OSHWA seems to be using controller/peripheral, fwiw The new terms are typically main/sub.

btw please don't mark comments as resolved as the discussion is ongoing :)

Reply:
Actually, the[ Revision 1.5 from May 22](https://downloadmirror.intel.com/27055/327432%20espi_base_specification%20R1-5.pdf) changed the language to Controller/Target:


• Updated the terms “master/slave” to “controller/target”
throughout the specification to align with the use of
inclusive language

But I think we should be able to follow this up with a quick PR to change everything over.

Reply:
Yes, I just saw your comments David!
Thanks for posting that.

Should we also change to CITO and COTI?
That would make sense, correct?

Reply:
> > The new terms are typically main/sub.
> 
> 
> 
> OSHWA seems to be using controller/peripheral, fwiw https://www.oshwa.org/a-resolution-to-redefine-spi-signal-names/
> 
> 
> 
> btw please don't mark comments as resolved as the discussion is ongoing :)

Sorry, I am using Biq bucket on my phone at the doctor's office while getting a blood draw.

I should put the phone down.